doc/using_xgboost.rst

```diff
@@ -0,0 +1,200 @@
+    parser.add_argument('--objective', type=str, default='reg:squarederror')
+    
+    # Sagemaker specific arguments. Defaults are set in the environment variables.
+    parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
+    parser.add_argument('--validation', type=str, default=os.environ['SM_CHANNEL_VALIDATION'])
+    
+    args = parser.parse_args()
+    
+    train_hp = {
+        'max_depth': args.max_depth,
+        'eta': args.eta,
+        'gamma': args.gamma,
+        'min_child_weight': args.min_child_weight,
+        'subsample': args.subsample,
+        'silent': args.silent,
+        'objective': args.objective
+    }
+    
+    dtrain = xgb.DMatrix(args.train)
+    dval = xgb.DMatrix(args.validation)
+    watchlist = [(dtrain, 'train'), (dval, 'validation')] if dval is not None else [(dtrain, 'train')]
+
+    callbacks = []
+    prev_checkpoint, n_iterations_prev_run = add_checkpointing(callbacks)
+    # If checkpoint is found then we reduce num_boost_round by previously run number of iterations
+    
+    bst = xgb.train(
+        params=train_hp,
+        dtrain=dtrain,
+        evals=watchlist,
+        num_boost_round=(args.num_round - n_iterations_prev_run),
+        xgb_model=prev_checkpoint,
+        callbacks=callbacks
+    )
+    
+    model_location = args.model_dir + '/xgboost-model'
+    pkl.dump(bst, open(model_location, 'wb'))
+    logging.info("Stored trained model at {}".format(model_location))
+
+In the training script, you can customize the inference behavior by implementing the follwing functions:
+* ``input_fn`` - how input data is handled.
+* ``predict_fn`` - how the model is invokedfunction, and how the response is returned ).
+* ``output_fn`` - How the response data is handled
+
+These functions are optional. If you want to use the default implementations, do not implement them in your training script.
+
+Create an Estimator
+-------------------
+After you create your training script, create an instance of the :class:`sagemaker.xgboost.XGBoost` estimator.
+Pass an IAM role that has the permissions necessary to run an Amazon SageMaker training job,
+the type and number of instances to use for the training job,
+and a dictionary of the hyperparameters to pass to the training script.
+
+.. code::
+
+    from sagemaker.session import s3_input
+    from sagemaker.xgboost.estimator import XGBoost
+
+    xgb_script_mode_estimator = XGBoost(
+        entry_point="abalone.py",
+        hyperparameters=hyperparameters,
+        image_name=container,
+        role=role, 
+        train_instance_count=1,
+        train_instance_type="ml.m5.2xlarge",
+        framework_version="0.90-1",
+        output_path="s3://{}/{}/{}/output".format(bucket, prefix, "xgboost-script-mode"),
+        train_use_spot_instances=train_use_spot_instances,
+        train_max_run=train_max_run,
+        train_max_wait=train_max_wait,
+        checkpoint_s3_uri=checkpoint_s3_uri
```

Comment:
do we need all of these? since this is just a basic example, I think only `entry_point`, `hyperparameters`, `role`, `train_instance_count`, `train_instance_type`, and `framework_version` are necessary